ocp_resources/datavolume.py

```diff
@@ -229,3 +240,52 @@ def _check_none_pending_status(self, failure_timeout=120):
         except TimeoutExpiredError:
             self.logger.error(f"{self.name} status is {sample}")
             raise
+
+    def is_garbage_collector_enabled_dv(self):
+        if self.exists:
+            dv_annotations = self.instance.metadata.get("annotations")
+            if dv_annotations:
+                return (
+                    dv_annotations.get("cdi.kubevirt.io/storage.deleteAfterCompletion")
+                    is not False
+                )
+        else:
+            if self.pvc.exists and self.pvc.instance.status.phase == "Bound":
+                return True
+            return False
+
+    def wait_for_dv_success(self, timeout=600, failure_timeout=120):
+        LOGGER.info(f"Wait DV success for {timeout} seconds")
+        self._check_none_pending_status(failure_timeout=failure_timeout)
+        self.pvc.wait_for_status(
+            status=PersistentVolumeClaim.Status.BOUND, timeout=timeout
+        )
+        # if garbage collector enabled, the DV deleted once succeeded
+        if (
+            CDIConfig.is_garbage_collector_enabled_cdi_config()
+            and self.is_garbage_collector_enabled_dv()
+        ):
+            try:
+                for sample in TimeoutSampler(
+                    sleep=1,
+                    wait_timeout=timeout,
+                    func=self._dv_phase,
+                ):
+                    # DV reach to success if the status is succeeded or if the DV does not exists
+                    if sample in [self.Status.SUCCEEDED, DV_DOES_NOT_EXISTS]:
+                        return
+            except TimeoutExpiredError:
+                LOGGER.error(f"Status of {self.kind} {self.name} is {sample}")
+            raise
```

Comment:
`raise` should be under `except TimeoutExpiredError`

Reply:
ack